### PR TITLE
Update Icons.ts with Farcaster Purple for SocialIcons

### PR DESCRIPTION
**Pull request by SamuelLHuber**
#### Description

This PR adds farcaster.xyz to the options in the SocialIcons menu.

SVG for the icon can be found here https://github.com/vrypan/farcaster-brand/blob/main/icons/icon-transparent/transparent-purple.svg

### CODE REVIEW

**Comment by delucis** [comment]
Hi @SamuelLHuber — thanks for the PR. Could you double check the SVG path you provided? I don’t see the icon showing up in the [preview of the docs](https://starlight-git-fork-samuellhuber-patch-1-astrodotbuild.vercel.app/guides/components/#icon):

icon grid space labelled “farcaster” but it is empty

**Comment by SamuelLHuber** [comment]
thank you for replying this quickly, appreciated.

this is the original SVG from the github source
```
<svg width="16" height="16" viewBox="0 0 1000 1000" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M257.778 155.556H742.222V844.445H671.111V528.889H670.414C662.554 441.677 589.258 373.333 500 373.333C410.742 373.333 337.446 441.677 329.586 528.889H328.889V844.445H257.778V155.556Z" fill="#855DCD"/>
<path d="M128.889 253.333L157.778 351.111H182.222V746.667C169.949 746.667 160 756.616 160 768.889V795.556H155.556C143.283 795.556 133.333 805.505 133.333 817.778V844.445H382.222V817.778C382.222 805.505 372.273 795.556 360 795.556H355.556V768.889C355.556 756.616 345.606 746.667 333.333 746.667H306.667V253.333H128.889Z" fill="#855DCD"/>
<path d="M675.556 746.667C663.283 746.667 653.333 756.616 653.333 768.889V795.556H648.889C636.616 795.556 626.667 805.505 626.667 817.778V844.445H875.556V817.778C875.556 805.505 865.606 795.556 853.333 795.556H848.889V768.889C848.889 756.616 838.94 746.667 826.667 746.667V351.111H851.111L880 253.333H702.222V746.667H675.556Z" fill="#855DCD"/>
</svg>
```

which matches the one in the commit, unsure if that helps.

it is working with that exact svg on a overriden SocialIcons Header i have deployed

**Comment by SamuelLHuber** [comment]
check thank you resized using svgviewer.dev and commiting the resized svg now

full svg for reference
```
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M6.187 3.733h11.627v16.533h-1.707V12.693h-0.017c-0.189 -2.093 -1.948 -3.733 -4.09 -3.733s-3.901 1.64 -4.09 3.733h-0.017v7.573h-1.707z" fill="#855DCD"/><path d="m3.093 6.08 0.693 2.347h0.587v9.493c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h-0.107c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h5.973v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533h-0.107v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533h-0.64V6.08z" fill="#855DCD"/><path d="M16.213 17.92c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h-0.107c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h5.973v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533h-0.107v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533V8.427h0.587L21.12 6.08H16.853v11.84z" fill="#855DCD"/></svg>
```

**Comment by SamuelLHuber** [comment]
shows up perfectly in preview now

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -64,6 +64,8 @@ export const BuiltInIcons = {
...
 	codePen:
 		'<path d="M23.5 7.5 12.5.2a1 1 0 0 0-1 0L.4 7.5a1 1 0 0 0-.5.8v7.4c0 .3.2.6.5.8l11 7.3c.3.3.7.3 1 0l11-7.3c.3-.2.5-.5.5-.8V8.3a1 1 0 0 0-.5-.8zM13 3l8.1 5.3-3.6 2.5-4.5-3V3zm-2 0v4.8l-4.5 3-3.6-2.5 8-5.3zm-9 7.3L4.7 12l-2.5 1.7v-3.4zM11 21l-8.1-5.3 3.6-2.5 4.5 3V21zm1-6.6L8.4 12 12 9.6l3.6 2.4-3.6 2.4zm1 6.6v-4.8l4.5-3 3.6 2.5-8 5.3zm9-7.3L19.3 12l2.5-1.7v3.4z"/>',
+	farcaster:
+		'<path d="M6.187 3.733h11.627v16.533h-1.707V12.693h-0.017c-0.189 -2.093 -1.948 -3.733 -4.09 -3.733s-3.901 1.64 -4.09 3.733h-0.017v7.573h-1.707z" fill="#855DCD"></path> <path d="m3.093 6.08 0.693 2.347h0.587v9.493c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h-0.107c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h5.973v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533h-0.107v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533h-0.64V6.08z" fill="#855DCD"></path> <path d="M16.213 17.92c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h-0.107c-0.295 0 -0.533 0.239 -0.533 0.533v0.64h5.973v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533h-0.107v-0.64c0 -0.295 -0.239 -0.533 -0.533 -0.533V8.427h0.587L21.12 6.08H16.853v11.84z" fill="#855DCD"></path>',
```
Comment (HiDeoo) [inline packages/starlight/components/Icons.ts:68]:
```suggestion
		'<path d="M6.187 3.733h11.627v16.533h-1.707v-7.573h-.017a4.107 4.107 0 0 0-8.18 0h-.017v7.573H6.186z"/><path d="m3.093 6.08.693 2.347h.587v9.493a.533.533 0 0 0-.533.533v.64h-.107a.533.533 0 0 0-.533.533v.64h5.973v-.64a.533.533 0 0 0-.533-.533h-.107v-.64A.533.533 0 0 0 8 17.92h-.64V6.08zM16.213 17.92a.533.533 0 0 0-.533.533v.64h-.107a.533.533 0 0 0-.533.533v.64h5.973v-.64a.533.533 0 0 0-.533-.533h-.107v-.64a.533.533 0 0 0-.533-.533V8.427h.587l.693-2.347h-4.267v11.84z"/>',
```

Thanks for the update. The SVG does indeed now properly shows up but would require a few more tiny changes:

- The hardcoded color should be removed.
- We can still optimize the resized SVG a bit using [SVGOMG](https://jakearchibald.github.io/svgomg/)

The suggested change I submitted contains this and renders as follows:

image

**Review by HiDeoo — APPROVED** [review]
Thanks for updating the pull request 🙌 

This is looking really great. I added a changeset file that will be used to describe the changes to users. This looks perfect to me.

Note that this PR won't be merged immediately but will be merged when the next release is made so that the documentation mentioning the new icon is only updated when the new icon is actually available.

**Pull request merged**